doc: Consolidate release notes before 0.19.0 (move-only)

maflcko opened the pull request.

Commit:
doc: Add missing release notes for 16383

MarcoFalke commented:
Also, add missing notes for #16383, follow up to https://github.com/bitcoin/bitcoin/pull/16690#discussion_r329287449

Commit:
doc: Consolidate release notes before 0.19.0

Can be reviewed with the git diff options:

--color-moved=dimmed-zebra --color-moved-ws=ignore-all-space

jonatack commented on doc/release-notes.md:
```diff
@@ -168,19 +168,32 @@ Low-level Changes section below.
   user (that is, before the descriptor is normalized for the
   `descriptor` field). (#15986)
 
+- `joinpsbts` will shuffle the order of the inputs and outputs of the resulting joined psbt.
```
Nit: use present tense here like the other notes do, e.g. s/will shuffle/now shuffles/

MarcoFalke replied:
I'd prefer to keep this move-only. It can be changed in the wiki by anyone, once they are moved: https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.19.0-Release-Notes-Draft

jonatack commented on doc/release-notes.md:
```diff
@@ -168,19 +168,32 @@ Low-level Changes section below.
   `descriptor` field). (#15986)
 
+- `joinpsbts` will shuffle the order of the inputs and outputs of the resulting joined psbt.
+  Previously inputs and outputs were added in the order that the PSBTs were provided which makes correlating inputs to outputs extremely easy.
```
Suggestion: "Previously, inputs and outputs were added in the order PSBTs were provided. This made it easy to correlate inputs to outputs, representing a privacy leak."

MarcoFalke replied:
I'd prefer to keep this move-only. It can be changed in the wiki by anyone, once they are moved: https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.19.0-Release-Notes-Draft

jonatack reviewed (commented):
A couple of suggestions below.

fanquake reviewed (approved):
ACK fa3d98426b35c0ece2853ce9fe46acd46f678ba3 - these can get massaged / nitted to death in the wiki.

laanwj commented:
Posthumous ACK.

jonatack commented:
Right, future release notes draft at:

https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.19.0-Release-Notes-Draft
